libtransmission/bandwidth.cc

```diff
@@ -288,97 +288,94 @@ void tr_bandwidthAllocate(tr_bandwidth* b, tr_direction dir, unsigned int period
+            if (now == 0)
             {
-                double current;
-                double desired;
-                double r;
-
-                if (now == 0)
-                {
-                    now = tr_time_msec();
-                }
-
-                current = tr_bandwidthGetRawSpeed_Bps(b, now, TR_DOWN);
-                desired = tr_bandwidthGetDesiredSpeed_Bps(b, TR_DOWN);
-                r = desired >= 1 ? current / desired : 0;
-
-                if (r > 1.0)
-                {
-                    byteCount = 0;
-                }
-                else if (r > 0.9)
-                {
-                    byteCount *= 0.8;
-                }
-                else if (r > 0.8)
-                {
-                    byteCount *= 0.9;
-                }
+                now = tr_time_msec();
             }
-        }
 
-        if (b->parent != nullptr && b->band[dir].honorParentLimits && byteCount > 0)
-        {
-            byteCount = bandwidthClamp(b->parent, now, dir, byteCount);
+            current = this->getRawSpeed_Bps(now, TR_DOWN);
+            desired = tr_bandwidthGetDesiredSpeed_Bps(this, TR_DOWN);
+            r = desired >= 1 ? current / desired : 0;
+
+            if (r > 1.0)
+            {
+                byteCount = 0;
+            }
+            else if (r > 0.9)
+            {
+                byteCount = static_cast<unsigned int>(byteCount * 0.8);
+            }
+            else if (r > 0.8)
+            {
+                byteCount = static_cast<unsigned int>(byteCount * 0.9);
+            }
         }
     }
 
+    if (this->parent != nullptr && this->band[dir].honorParentLimits && byteCount > 0)
+    {
+        byteCount = this->parent->clamp(now, dir, byteCount);
+    }
+
     return byteCount;
 }
 
-unsigned int tr_bandwidthClamp(tr_bandwidth const* b, tr_direction dir, unsigned int byteCount)
+unsigned int tr_bandwidth::clamp(tr_direction dir, unsigned int byteCount) const
 {
-    return bandwidthClamp(b, 0, dir, byteCount);
+    return this->clamp(0, dir, byteCount);
 }
 
-unsigned int tr_bandwidthGetRawSpeed_Bps(tr_bandwidth const* b, uint64_t const now, tr_direction const dir)
+unsigned int tr_bandwidth::getRawSpeed_Bps(uint64_t const now, tr_direction const dir) const
 {
-    TR_ASSERT(tr_isBandwidth(b));
+    TR_ASSERT(tr_isBandwidth(this));
     TR_ASSERT(tr_isDirection(dir));
 
-    return getSpeed_Bps(&b->band[dir].raw, HISTORY_MSEC, now);
+    return getSpeed_Bps(&this->band[dir].raw, HISTORY_MSEC, now);
 }
 
-unsigned int tr_bandwidthGetPieceSpeed_Bps(tr_bandwidth const* b, uint64_t const now, tr_direction const dir)
+unsigned int tr_bandwidth::getPieceSpeed_Bps(uint64_t const now, tr_direction const dir) const
 {
-    TR_ASSERT(tr_isBandwidth(b));
+    TR_ASSERT(tr_isBandwidth(this));
     TR_ASSERT(tr_isDirection(dir));
 
-    return getSpeed_Bps(&b->band[dir].piece, HISTORY_MSEC, now);
+    return getSpeed_Bps(&this->band[dir].piece, HISTORY_MSEC, now);
 }
```

Comment:
Likewise, these are all effectively one-liners. They could be moved into the .h file IMO